src/components/templates/PosterHall/PosterHall.tsx

```diff
@@ -31,15 +53,19 @@ export const PosterHall: React.FC<PosterHallProps> = ({ venue }) => {
     setLiveFilter,
   } = usePosters(venue.id);
 
+  const posterVenuesPersonalized = posterVenues.map(
+    prepareForMyPosters(userPosterIds)
+  );
+
   const renderedPosterPreviews = useMemo(() => {
-    return posterVenues.map((posterVenue) => (
+    return posterVenuesPersonalized.map((posterVenue) => (
       <PosterPreview
         key={posterVenue.id}
-        posterVenue={posterVenue}
         enterVenue={enterVenue}
+        personalizedPoster={posterVenue}
       />
     ));
-  }, [posterVenues]);
+  }, [posterVenuesPersonalized]);
```

Comment:
Since all of the changes here related to `userPosterIds`, which is just extracted from `userWithId` which comes from the `useUser` hook, there doesn't seem to be any reason to be doing all of this processing at this 'out leve', or changing things here.

I would just be reverting all of these changes back to how they were.

Reply:
thanks! reverted